src/rt/test/systematic.h

```diff
@@ -503,51 +504,14 @@ namespace Systematic
 
 #elif defined(CI_BUILD) && defined(USE_EXECINFO)
   static std::mutex mutx;
-  static std::condition_variable cv;
   static void* stack_frames = nullptr;
   static int n_frames = 0;
   static std::string systematic_id = "";
 
-  inline static void signal_handler(int sig, siginfo_t*, void*)
-  {
-    static std::atomic<bool> run_already = false;
-
-    if (run_already)
-      abort();
-    run_already = true;
-
-    auto str = strsignal(sig);
-
-    // We're ignoring the result of write, as there's not much we can do if it
-    // fails. We're about to crash anyway.
-    auto s1 = write(1, str, strlen(str));
-    auto s2 = write(1, "\n", 1);
-    UNUSED(s1 + s2);
-
-    constexpr size_t max_stack_frames = 64;
-    void* frames[max_stack_frames];
-    {
-      std::lock_guard<std::mutex> lock(mutx);
-      n_frames = backtrace(frames, max_stack_frames);
-      stack_frames = frames;
-      systematic_id = get_systematic_id();
-      cv.notify_one();
-    }
-    while (true)
-    {
-      sleep(1000);
-    }
-  }
 
   inline static void crash_dump()
   {
     std::unique_lock<std::mutex> lock(mutx);
```

Comment:
Do you still need this mutex if it is only ever called via ThreadPing?

Reply:
That lock is protecting the `stack` stuff.  As we only do this once, it should be fine without.  If we want to output repeatedly, then we need a lock so that the backtrace isn't raced on.  I'll add some comments.

Reply:
Sorry, I see that it is still needed to synchronize with signal_handler.